src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Android.cs

```diff
@@ -46,7 +46,7 @@ internal static partial class CertificateValidationPal
             bool retrieveChainCertificates,
             ref X509Chain? chain)
         {
-            SafeSslHandle sslContext = ((SafeDeleteSslContext)securityContext).SslContext;
+            SafeSslHandle? sslContext = ((SafeDeleteSslContext)securityContext)?.SslContext;
```

Comment:
this shouldn't happen given that `securityContext` isn't marked as nullable, which suggest to me something higher up the call chain is doing something unexpected.

Reply:
The underlaying handle _is_ nullable.
```
        private SafeFreeCredentials? _credentialsHandle;
        private SafeDeleteSslContext? _securityContext;
```

But is seems like it is like this for long time
https://github.com/dotnet/runtime/blob/3e5517beb897faf4592d23f036446561da1e5c23/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Protocol.cs#L254

Making the parameter nullable make sense to me. I think it will be null when we try to select client certificate before the handshake. cc: @rzikm

Reply:
The nullability annotations seem to be inconsistent across platforms:
- in the OSX version there's no `?` but there's a null check: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.OSX.cs#L55
- in the Unix version there's a `?` and there's a null check: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Unix.cs#L31
- in the Windows version there's a `?` and there's a null check: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs#L34

It seems that the `_securityContext` field in `SslStream` isn't properly initialized by the time `SelectClientCertificate` is called and it passes down the security context without checking if it's null or not and it uses the null forgiving operator. I'll investigate more and maybe I'll find a better fix than adding the null check.

Reply:
Ok thanks, that sounds like indeed we're expecting a null `securityContext` here and should make it consistent across platforms.